### ISSUE TITLE
App crash when going back from nested navigation

### ISSUE THREAD

**Issue by bengy** [rive-app/rive-react-native#212]
### Description

We have a setup in our app with Rive and React-Navigation where we navigate from our root stack into one that has a screen with a Rive animation. If we then navigate into another substack and then navigate back two times the app crashes with the following error:
```
Unhandled SoftException
FATAL EXCEPTION: main
Process: com.rivenesteddoublerelease, PID: 30266
java.lang.IllegalArgumentException: Failed requirement.
	at app.rive.runtime.kotlin.core.NativeObject.release(NativeObject.kt:77)
	at app.rive.runtime.kotlin.core.NativeObject.dispose(NativeObject.kt:102)
	at app.rive.runtime.kotlin.core.NativeObject.release(NativeObject.kt:80)
	at app.rive.runtime.kotlin.core.File.release(File.kt:116)
	at app.rive.runtime.kotlin.controllers.RiveFileController.setFile(RiveFileController.kt:90)
	at app.rive.runtime.kotlin.controllers.RiveFileController.release(RiveFileController.kt:720)
	at app.rive.runtime.kotlin.renderers.Renderer.disposeDependencies(Renderer.kt:225)
	at app.rive.runtime.kotlin.RiveArtboardRenderer.disposeDependencies(RiveArtboardRenderer.kt:341)
	at app.rive.runtime.kotlin.renderers.Renderer.cppDelete(Native Method)
	at app.rive.runtime.kotlin.renderers.Renderer.delete(Renderer.kt:212)
	at app.rive.runtime.kotlin.RiveTextureView.onDetachedFromWindow(RiveTextureView.kt:93)
	at app.rive.runtime.kotlin.RiveAnimationView.onDetachedFromWindow(RiveAnimationView.kt:873)
	at android.view.View.dispatchDetachedFromWindow(View.java:22051)
	at android.view.ViewGroup.clearDisappearingChildren(ViewGroup.java:7114)
	at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3981)
	at android.view.ViewGroup.clearDisappearingChildren(ViewGroup.java:7114)
	at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3981)
	at android.view.ViewGroup.clearDisappearingChildren(ViewGroup.java:7114)
	at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3981)
	at android.view.ViewGroup.clearDisappearingChildren(ViewGroup.java:7114)
	at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3981)
	at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3979)
	at android.view.ViewGroup.clearDisappearingChildren(ViewGroup.java:7114)
	at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3981)
	at android.view.ViewGroup.clearDisappearingChildren(ViewGroup.java:7114)
	at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3981)
	at android.view.ViewGroup.clearDisappearingChildren(ViewGroup.java:7114)
	at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3981)
	at android.view.ViewGroup.clearDisappearingChildren(ViewGroup.java:7114)
	at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3981)
	at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3979)
	at android.view.ViewGroup.endViewTransition(ViewGroup.java:7225)
	at com.swmansion.rnscreens.ScreenStack.endViewTransition(ScreenStack.kt:53)
	at androidx.fragment.app.DefaultSpecialEffectsController$4$1.run(DefaultSpecialEffectsController.java:258)
	at android.os.Handler.handleCallback(Handler.java:958)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:205)
	at android.os.Looper.loop(Looper.java:294)
	at android.app.ActivityThread.main(ActivityThread.java:8194)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)

```

### Provide a Repro
[Minimal Reproduction example](https://github.com/bengy/RiveNestedDoubleReleaseRepro)
Has some boilerplate from react-native init but the relevant parts is in `App.tsx`
```import Rive from 'rive-react-native';
import React from 'react';
import {Button, SafeAreaView, Text, View} from 'react-native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import {NavigationContainer, useNavigation} from '@react-navigation/native';

function OtherScreen(): React.JSX.Element {
  const navigation = useNavigation();
  return (
    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
      <Text>Other Nested Stack</Text>
      <Button onPress={() => navigation.goBack()} title="3. Go back" />
    </View>
  );
}

const OtherStack = createNativeStackNavigator();
function OtherNestedSTack(): React.JSX.Element {
  return (
    <OtherStack.Navigator>
      <Stack.Screen name="Other" component={OtherScreen} />
    </OtherStack.Navigator>
  );
}

function HomeScreen(): React.JSX.Element {
  const navigation = useNavigation();
  return (
    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
      <Text>Home Screen</Text>
      <Button
        onPress={() => navigation.navigate('RiveNestedStack', {screen: 'Rive'})}
        title="1. Navigate to nested rive"
      />
    </View>
  );
}

function RiveScreen(): React.JSX.Element {
  const navigation = useNavigation();
  return (
    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
      <Rive
        url="https://public.rive.app/community/runtime-files/2195-4346-avatar-pack-use-case.riv"
        artboardName="Avatar 1"
        stateMachineName="avatar"
        style={{width: 400, height: 400}}
      />
      <Button
        onPress={() =>
          navigation.navigate('OtherNestedStack', {screen: 'Other'})
        }
        title="2. Navigate to other nested"
      />
      <Button onPress={() => navigation.goBack()} title="5. Go back" />
    </View>
  );
}

const RiveStack = createNativeStackNavigator();
function RiveNestedStack(): React.JSX.Element {
  return (
    <RiveStack.Navigator>
      <Stack.Screen name="Rive" component={RiveScreen} />
    </RiveStack.Navigator>
  );
}

const Stack = createNativeStackNavigator();

function App(): React.JSX.Element {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen name="Home" component={HomeScreen} />
        <Stack.Screen name="RiveNestedStack" component={RiveNestedStack} />
        <Stack.Screen name="OtherNestedStack" component={OtherNestedSTack} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}
```

#### Source `.riv`/`.rev` file
Every rive animation we tested.
Also the public one from the example

### Expected behavior
Navigating back from the nested screen works if we do not navigate to another nested screen in between.

#### Additional context
In our App we use version "6.1.1" but the reproduction example uses "6.2.0"

**Comment by bengy** [comment]
> @bengy Do you still experience this if you downgrade `rive-android` runtime to `v8.4.0`

Yes the issue is still there with the older runtime.

**Comment by HayesGordon** [comment]
Hi all, appologies for the delay in this issue. This should be resolved in the latest v6.2.1 - please reopen if you encounter any issues.

**Comment by rickyrombo** [comment]
Hey @HayesGordon , I'm still running into issues on v6.2.1:

The repro sounds identical - navigate to a substack, back, and then to another substack and the app crashes.

```
02-09 11:32:00.899 27316 27316 E AndroidRuntime: java.lang.IllegalArgumentException: Failed requirement.
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at app.rive.runtime.kotlin.core.NativeObject.release(NativeObject.kt:77)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at app.rive.runtime.kotlin.RiveViewLifecycleObserver.onDestroy(RiveAnimationView.kt:1051)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at androidx.lifecycle.DefaultLifecycleObserverAdapter.onStateChanged(DefaultLifecycleObserverAdapter.kt:29)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.kt:314)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at androidx.lifecycle.LifecycleRegistry.backwardPass(LifecycleRegistry.kt:266)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.kt:283)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.kt:136)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.kt:119)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at androidx.fragment.app.FragmentViewLifecycleOwner.handleLifecycleEvent(FragmentViewLifecycleOwner.java:88)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at androidx.fragment.app.Fragment.performDestroyView(Fragment.java:3196)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at androidx.fragment.app.FragmentStateManager.destroyFragmentView(FragmentStateManager.java:742)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:346)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at androidx.fragment.app.SpecialEffectsController$FragmentStateManagerOperation.complete(SpecialEffectsController.java:745)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at androidx.fragment.app.SpecialEffectsController$Operation.completeSpecialEffect(SpecialEffectsController.java:672)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at androidx.fragment.app.DefaultSpecialEffectsController$SpecialEffectsInfo.completeSpecialEffect(DefaultSpecialEffectsController.java:776)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at androidx.fragment.app.DefaultSpecialEffectsController$4$1.run(DefaultSpecialEffectsController.java:259)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:883)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:100)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:214)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7356)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
02-09 11:32:00.899 27316 27316 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
```
